src/lib/operators/get_table.cpp

```diff
@@ -239,8 +240,35 @@ std::shared_ptr<const Table> GetTable::_on_execute() {
     ++output_chunks_iter;
   }
 
+  // Check if a table index is irrelevant because of pruning.
+  const auto all_indexed_segments_pruned = [&](const auto& table_index) {
+    // Check if indexed ColumnID has been pruned.
+    const auto indexed_column_id = table_index->get_indexed_column_id();
+    if (std::find(_pruned_column_ids.cbegin(), _pruned_column_ids.cend(), indexed_column_id) !=
+        _pruned_column_ids.cend()) {
+      return true;
+    }
+
+    const auto indexed_chunk_ids = table_index->get_indexed_chunk_ids();
+
+    // Early out if index is empty.
+    if (indexed_chunk_ids.empty()) {
```

Comment:
It's not possible to create an empty index, but it is possible to remove all indexed chunks from an index.

Reply:
Hmm, shouldn't we remove the index in this case?

I am fine with the current solution. Just want to make sure we are introducing any confusion. This would lead to an "indexed table" where nothing is indexed.

Reply:
Hm, definitely worth discussing, but wouldn't this need to be done by the table itself?
I don't know if this should be in the scope of this PR.

Reply:
Ahh, wait. Sorry. I was thinking we are removing all chunks from the physical index. But here, we are at the get table operator. Sorry.